<a href="https://colab.research.google.com/github/yuktha-projects/yuktha-projects/blob/main/Penguins_Sample_Data_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Pengiuns Sample Data Analysis**

* We have to use Pythin, NumPy and Pandas for analyzing the Physical Charecteristics of penguin





* Import Necessary Libraries


Mandatory omes:


In [2]:
import numpy as np
import pandas as pd

# for getting sample dataset
import seaborn as sns

Configurational Library

In [3]:
import warnings
warnings.filterwarnings('ignore')


to see the names of sample dataset available in seanborn library

In [4]:
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

Loaded the data

In [6]:
penguins = sns.load_dataset('penguins')

**Data Inspection**


Column analysis

In [7]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex'],
      dtype='object')

Shape analysis

In [14]:
penguins.shape
a = penguins.shape
a

(344, 7)

In [15]:
a = penguins.shape
print(f'The number of rows are {a[0]} and columns are {a[1]}')  # to present well

The number of rows are 344 and columns are 7


View the Data

In [17]:
penguins.head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   

   body_mass_g     sex  
0       3750.0    Male  
1       3800.0  Female  
2       3250.0  Female  
3          NaN     NaN  
4       3450.0  Female

Insight:
* There are some null values in our dataset, we can either drop them or fill them

**Data types of columns analysis**

In [18]:
penguins.dtypes

species               object
island                object
bill_length_mm       float64
bill_depth_mm        float64
flipper_length_mm    float64
body_mass_g          float64
sex                   object
dtype: object

**Informaton of data**

In [19]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


Check the total null values

In [21]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

In [23]:
# Here we are going to drop null values cuz it might give you vague values when you fill the null values using mode for the gender.

penguins.dropna(inplace = True)

Recheck the shape of the sata

In [25]:
a = penguins.shape
print(f'The number of rows are {a[0]} and columns are {a[1]}')

The number of rows are 333 and columns are 7


To check **Numerical Sats Summary**

In [26]:
penguins.describe().T

count         mean         std     min     25%     50%  \
bill_length_mm     333.0    43.992793    5.468668    32.1    39.5    44.5   
bill_depth_mm      333.0    17.164865    1.969235    13.1    15.6    17.3   
flipper_length_mm  333.0   200.966967   14.015765   172.0   190.0   197.0   
body_mass_g        333.0  4207.057057  805.215802  2700.0  3550.0  4050.0   

                      75%     max  
bill_length_mm       48.6    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4775.0  6300.0

INSIGHTS:
* **Mean Column**
* On an average the bill length is 43.99
* Avg bill depth: 17.16
* Avg flipper length: 300.96
* Avg body mass: 4207.05

** Min column:**
minimun..bill length a penguin having is 32.1, etc


**Categorical Data Summary

In [31]:
penguins.describe(include='O').T

count unique     top freq
species   333      3  Adelie  146
island    333      3  Biscoe  163
sex       333      2    Male  168

Feature engineering (creating a new column based on the existing columns)

Add a new column for body mass in kilograms

In [37]:
# Defining the column name and adding new column.
penguins['body_mass_kg'] = penguins['body_mass_g']/1000
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
338  Gentoo     Biscoe            47.2           13.7              214.0   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  body_mass_kg  
0         3750.0    Male         3.750  
1         3800.0  Female         3.800  
2         3250.0  Female         3.250  
4         3450.0  Female         3.450  
5         3650.0    Male         3.650  
..           ...     ...           ...  
338       4925.0  Female         4.925  
340       4850.0  Female         4.850  
341       5750.0    Male         5.750  
342       5200.0  Female         5.200  
343       5400.0    Male         5.400  

[333 rows x 8 columns]

## **Exploratory Data Analysis**
Number of penguins per species

In [47]:
penguins['species'].value_counts() #value_counts is not an attribute it is a method hence close the brackets

species
Adelie       146
Gentoo       119
Chinstrap     68
Name: count, dtype: int64

Get the propostion of penguins per species


In [49]:
penguins['species'].value_counts(normalize=True)*100   #

species
Adelie       43.843844
Gentoo       35.735736
Chinstrap    20.420420
Name: proportion, dtype: float64

**Lets find the average measurement per species**
Create a table to get an average mesurement of bill length, depth and flipper lenght


In [58]:
avgSpeciesMeasurement = penguins.groupby('species')[['bill_length_mm', 'bill_depth_mm' , 'flipper_length_mm', 'body_mass_g']].mean()
avgSpeciesMeasurement

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
species                                                                 
Adelie          38.823973      18.347260         190.102740  3706.164384
Chinstrap       48.833824      18.420588         195.823529  3733.088235
Gentoo          47.568067      14.996639         217.235294  5092.436975

**Insights:**
* On an Average Chinstrap species are having bigger bill length and depth compared to other 2 species.
* on an average Gentoo species are having bigger flipper length and body mass



** Gender Based Analysis **


Which gender is having higher Body mass?


In [60]:
avgBodyMass_Male_Female = penguins.groupby('sex')['body_mass_g'].mean()
avgBodyMass_Male_Female

sex
Female    3862.272727
Male      4545.684524
Name: body_mass_g, dtype: float64

Insight:
* On an average male penguins wieghts more than Female

**Island-wise distribution**

Which species are found on which Island?


In [71]:
TotalSpecies_Island = penguins.groupby('species')['island'].value_counts() #value_counts gives me for individual species how many penguins are there in thet particular island
TotalSpecies_Island

species    island   
Adelie     Dream         55
           Torgersen     47
           Biscoe        44
Chinstrap  Dream         68
Gentoo     Biscoe       119
Name: count, dtype: int64

Insight:

*

**NumPy**

* Find the average penguin body mass
* Extract the penguins which are heavier than the average body mass

In [78]:
avgBody_mass = penguins.body_mass_g.mean()
avgBody_mass

np.float64(4207.057057057057)

In [81]:
avgBody_mass = penguins.body_mass_g.mean()
avgBody_mass = round(avgBody_mass,2)
avgBody_mass

np.float64(4207.06)

In [82]:
penguins[penguins['body_mass_g']>avgBody_mass].head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7   Adelie  Torgersen            39.2           19.6              195.0   
14  Adelie  Torgersen            34.6           21.1              198.0   
17  Adelie  Torgersen            42.5           20.7              197.0   
39  Adelie      Dream            39.8           19.1              184.0   
43  Adelie      Dream            44.1           19.7              196.0   

    body_mass_g   sex  body_mass_kg  
7        4675.0  Male         4.675  
14       4400.0  Male         4.400  
17       4500.0  Male         4.500  
39       4650.0  Male         4.650  
43       4400.0  Male         4.400

**Create a column (Size Category)**

* Classify the group into 3 categories
       

        *   Light weight: < 4000 g
        *   Medium weight: `4000-5000 g`
        *   Heavy Weight: > 5000 g


        

     

In [84]:
penguins['sizeCategory'] = np.where(
    penguins['body_mass_g'] < 4000 , 'LightWeight', 'Heavy'
)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
338  Gentoo     Biscoe            47.2           13.7              214.0   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  body_mass_kg sizeCategory  
0         3750.0    Male         3.750  LightWeight  
1         3800.0  Female         3.800  LightWeight  
2         3250.0  Female         3.250  LightWeight  
4         3450.0  Female         3.450  LightWeight  
5         3650.0    Male         3.650  LightWeight  
..           ...     ...           ...          ...  
338       4925.0  Female         4.925        Heavy  
340       4850.0  Female         4.850        Heavy  
341       5750.0    Male         5.750        Heavy  
342       5200.0  Female         5.200        Heavy  
343       5400.0    Male         5.400        Heavy  

[333 rows x 9 columns]

In [88]:
penguins['sizeCategory'] = np.where(
    penguins['body_mass_g'] < 4000 , 'LightWeight',
    np.where(penguins['body_mass_g'] <= 5000, 'Medium', 'Heavy')
)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
338  Gentoo     Biscoe            47.2           13.7              214.0   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  body_mass_kg sizeCategory  
0         3750.0    Male         3.750  LightWeight  
1         3800.0  Female         3.800  LightWeight  
2         3250.0  Female         3.250  LightWeight  
4         3450.0  Female         3.450  LightWeight  
5         3650.0    Male         3.650  LightWeight  
..           ...     ...           ...          ...  
338       4925.0  Female         4.925       Medium  
340       4850.0  Female         4.850       Medium  
341       5750.0    Male         5.750        Heavy  
342       5200.0  Female         5.200        Heavy  
343       5400.0    Male         5.400        Heavy  

[333 rows x 9 columns]

In [89]:
penguins[['species','body_mass_g','sizeCategory']]

species  body_mass_g sizeCategory
0    Adelie       3750.0  LightWeight
1    Adelie       3800.0  LightWeight
2    Adelie       3250.0  LightWeight
4    Adelie       3450.0  LightWeight
5    Adelie       3650.0  LightWeight
..      ...          ...          ...
338  Gentoo       4925.0       Medium
340  Gentoo       4850.0       Medium
341  Gentoo       5750.0        Heavy
342  Gentoo       5200.0        Heavy
343  Gentoo       5400.0        Heavy

[333 rows x 3 columns]

In [90]:
penguins.sizeCategory.value_counts()

sizeCategory
LightWeight    161
Medium         111
Heavy           61
Name: count, dtype: int64